src/Compilers/Core/Portable/Syntax/InternalSyntax/SyntaxNodeCache.cs

```diff
@@ -278,7 +278,9 @@ private static int GetCacheHash(int kind, GreenNode.NodeFlags flags, GreenNode?
         {
             int code = (int)(flags) ^ kind;
             // the only child is never null
-            code = Hash.Combine(System.Runtime.CompilerServices.RuntimeHelpers.GetHashCode(child1), code);
+            code = Hash.Combine(
+                child1 is object ? System.Runtime.CompilerServices.RuntimeHelpers.GetHashCode(child1) : 0,
```

Comment:
I'm surprised. I would expect `RuntimeHelpers.GetHashCode` to handle `null`.
It's even [tested](https://github.com/dotnet/runtime/blob/4f9ae42d861fcb4be2fcd5d3d55d5f227d30e723/src/libraries/System.Runtime/tests/System/Runtime/CompilerServices/RuntimeHelpersTests.cs#L35).

Reply:
@stephentoub can comment here but I verified that the annotation currently prohibits `null` values.

Reply:
This has been fixed for .NET 5:
https://github.com/dotnet/runtime/blob/a98abf4db5845c072384a436ae7411f8c6439863/src/libraries/System.Runtime/ref/System.Runtime.cs#L7276
https://github.com/dotnet/runtime/blob/a98abf4db5845c072384a436ae7411f8c6439863/src/coreclr/src/System.Private.CoreLib/src/System/Runtime/CompilerServices/RuntimeHelpers.CoreCLR.cs#L110
https://github.com/dotnet/runtime/blob/a98abf4db5845c072384a436ae7411f8c6439863/src/mono/netcore/System.Private.CoreLib/src/System/Runtime/CompilerServices/RuntimeHelpers.Mono.cs#L25

Reply:
@jaredpar I would prefer to use `!` here since we will eventually have an analyzer to aid in the removal of unnecessary `!`, but we can't do the same as easily if you change the code flow.

Reply:
> we will eventually have an analyzer to aid in the removal of unnecessary `!`

@sharwell Sounds great, but just to let you know, I'm not sure if we have sufficient compiler APIs to achieve that. Here's one API we're potentially missing: https://github.com/dotnet/roslyn/issues/35412

Reply:
@jcouv The diagnostic would be implemented directly by the compiler, similar to CS8019 used for unused imports.

Reply:
@sharwell That is certainly a possible design, and I think it wouldn't be too hard (the compiler has all the relevant information, ie type before and after suppression, when we process suppressions).

Tagging @jaredpar for the compiler ask and scheduling: we would need to produce hidden diagnostics for unnecessary suppressions.